From 367d5377a2b9f6e1df9034e382170cada9d9992e Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Tue, 7 Apr 2026 16:45:57 +0800
Subject: [PATCH 3/3] FROMLIST: skcms: Disable musttail on MIPS, PowerPC, and
 LoongArch with GCC

The GCC issue linked for RISC-V also affects MIPS, PowerPC, and LoongArch
(I've edited the GCC ticket to mention them as well).

Change-Id: I5ac392142e58178300e3882baeb70695e31c8db8

Link: https://skia-review.googlesource.com/c/skcms/+/1203676
Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 Source/ThirdParty/skia/modules/skcms/src/skcms_internals.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/Source/ThirdParty/skia/modules/skcms/src/skcms_internals.h b/Source/ThirdParty/skia/modules/skcms/src/skcms_internals.h
index 02aa7b77a58..274471935fc 100644
--- a/Source/ThirdParty/skia/modules/skcms/src/skcms_internals.h
+++ b/Source/ThirdParty/skia/modules/skcms/src/skcms_internals.h
@@ -59,7 +59,10 @@ extern "C" {
     #elif defined(__GNUC__) && !defined(SKCMS_HAS_MUSTTAIL)
         // GCC on riscv64 does not support our tail call functions
         // cf. https://gcc.gnu.org/bugzilla/show_bug.cgi?id=121784
-        #if __has_cpp_attribute(clang::musttail) && !defined(__riscv)
+        #if __has_cpp_attribute(clang::musttail) && !defined(__riscv) \
+                                                 && !defined(__mips__) \
+                                                 && !defined(__powerpc__) \
+                                                 && !defined(__loongarch__)
             #define SKCMS_HAS_MUSTTAIL 1
         #else
             #define SKCMS_HAS_MUSTTAIL 0
-- 
2.54.0

